common/src/main/java/io/netty/util/concurrent/PromiseNotifier.java

```diff
@@ -61,6 +61,57 @@ public PromiseNotifier(boolean logNotifyFailure, Promise<? super V>... promises)
         this.logNotifyFailure = logNotifyFailure;
     }
 
+    /**
+     * Fuse the {@link Future} and {@link Promise}. This means that if the {@link Future} completes the {@link Promise}
+     * will be notified. That said cancellation is propagated both ways. This means if the {@link Future} is cancelled
+     * the {@link Promise} is cancelled as well and vise-versa.
+     *
+     * @param future    the {@link Future} which will be used to listen to for notifying the {@link Promise}.
+     * @param promise   the {@link Promise} which will be notified
+     * @param <V>       the type of the value.
+     * @param <F>       the type of the {@link Future}
+     * @return          the passed in {@link Future}
+     */
+    public static <V, F extends Future<V>> F fuse(final F future, final Promise<? super V> promise) {
```

Comment:
I am open to a better name :)

Reply:
I think "fuse" is only bad in that it has two meanings; fuse as in fusion, or fuse as in an electrical fuse.
A few alternatives I can think of: link, entangle (oof, probably too niche or technical), correlate.

Reply:
@chrisvest link is good... I changed it to `link`. Thanks for the suggestions!